docs/contributor-guide/how-to/common-rule-scenarios.md

```diff
@@ -0,0 +1,205 @@
+# Implement common rule scenarios
+
+This page documents the most common scenarios encoutered when developing a rule
+for `sonarwhal`. If there's something that you want to do that is not
+documented here, please [open an issue][new issue] so we can help you.
+
+## Change feedback based on browser support
+
+Users can tell `sonarwhal` what browsers are important for them via the
+[`browserslist` property in `.sonarwhalrc`][browserconfiguration] or in
+the `package.json` file. You can have access to the list, and thus modify
+the feedback of your rule, via the property `context.targetedBrowsers`.
+
+<!-- eslint-disable no-unused-vars -->
+
+```js
+const validate = (fetchEnd) => {
+    if (!context.targetedBrowsers.includes('Edge 14')) {
+        return;
+    }
+
+    // Your validations
+};
+```
+
+## Reporting an error if not run
+
+Sometimes what a rule checks is mandatory, and if it does not have
+the chance to test it, it should fail. These are the types of rules
+that enforce certain things to be used in a certain way, and if
+included, in order for the rule to pass, the expectation should be
+that the thing the rule checks for should exist and be valid/used
+correctly. Examples here are the rules that check for different fields
+of the manifest file. They should not pass if, for example, the web
```

Comment:
This is not a good example since the manifest related rules don't actually do anything if the manifest doesn't exist.